packages/adminrouter/extra/src/README.md

```diff
@@ -264,28 +264,178 @@ it, but it does not seem to be justified. For example:
+Mesos frameworks. For example - let's deploy a SchmetterlingDB framework into
+`/group1/group2/schmetterlingA`. It will be accessible using its ID (let's assume
+that Mesos assigned it a framework ID of 819aed93-4143-4291-8ced-5afb5c726803-0000):
 ```
-"labels": {
-    "DCOS_SERVICE_NAME": "service-name",
-    "DCOS_SERVICE_PORT_INDEX": "0",
-    "DCOS_SERVICE_SCHEME": "http"
-  }
+/service/819aed93-4143-4291-8ced-5afb5c726803-0000/
+```
+and using its name:
+```
+/service/group1/group2/schmetterlingA/
 ```
 
-When your container/task has its own IP (typically when running in a virtual network), `http://<dcos-cluster>/service/service-name` would be forwarded to the container/task's IP using one of the ports in the port mapping or port definition when USER networking is enabled or one of the discovery ports otherwise. When your task/container is mapped to the host, it would be forwarded to the host running the task using the one of the ports allocated to the task. The chosen port is defined by the DCOS_SERVICE_PORT_INDEX label.
-
-In order for the forwarding to work reliably across task failures, we recommend co-locating the endpoints with the task. This way, if the task is restarted on a potentially other host and with different ports, Admin Router will pick up the new labels and update the routing. NOTE: Due to caching there might be an up to 30-second delay until the new routing is working.
-
-We would recommend having only a single task setting these labels for a given `service-name`.
-In the case of multiple task instances with the same `service-name` label, Admin Router will pick one of the tasks instances deterministically, but this might make debugging issues more difficult.
+The `/service` endpoint supports WebSockets, by default disables all NGINX
+buffering and all redirects from services are rewritten in order to support
+`/service` prefix clients are using.
+
+### Intended use
+It is very important to remember that `/service` endpoint was designed to work
+with root Marathon only. Tasks launched by MarathonOnMarathon instances will not
+be reachable via it.
+
+In the case of Mesos frameworks, this becomes a bit more complex - all the
+frameworks are visible in Mesos' `/state-summary` endpoint output so in theory
+it is possible to access a framework launched using MoM or by hand.
+Unfortunately, Mesos as of now does not enforce framework name uniqueness and
+thus it is possible to launch two frameworks with the same name. AR in such
+case will route in a deterministic manner. Root Marathon by default enforces
+unique names for all the tasks and frameworks it has under control.
+
+Please check the [Limitations](#limitations) section for more details.
+
+### Operation
+Under the hood, the `/service` endpoint follows non-trivial algorithm in order to
+resolve request path into IP + Port pair.
+
+#### Recursive resolving
+There is no way to tell which part of the path is namespace+service name and
```

Comment:
This is hard or impossible to understand as of the word "namespace" -- we should add clarity by proper term definition.